docs/tech-notes/index-backfill.md

```diff
@@ -0,0 +1,156 @@
+# Index backfills
+
+This document explains why index backfills using historical reads and retry
+transactions are always correct.
+
+Index backfills work as follows:
+
+1. Wait until index converges to write/delete state. This means that every
+   write and delete to the primary table gets reflected in the index.
+2. Fix an MVCC timestamp, `tScan`. This timestamp will be used to perform scans
+   in the fast path.
```

Comment:
Since the scans are performed at a historical timestamp they will bypass the raft execution queue.